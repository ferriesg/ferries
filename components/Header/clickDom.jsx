import React, { useState } from "react";

export default function clickDom() {
  const [buttonTitle, setButtonTitle] = useState("on");
  return (
    <button
      className="button"
      onClick={() => {
        setButtonTitle("off");
      }}
    >
      {buttonTitle}
    </button>
  );
}
