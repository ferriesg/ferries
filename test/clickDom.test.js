import enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import clickDom from "../components/Header/clickDom";
import jsdom from "jsdom";
enzyme.configure({ adapter: new Adapter() });

const { JSDOM } = jsdom;
const { window } = new JSDOM("");
const { document } = new JSDOM(``).window;
global.document = document;
global.window = window;

test("click dom test", () => {
  const currentDom = enzyme.mount(<clickDom />);
  // console.log(currentDom.debug());
  expect(currentDom.debug()).toMatchSnapshot();
  // console.log(expect(currentDom.exists(".button")));
  // const buttonEle = currentDom.getByRole("button",{ name: 'on' });
  // console.log(buttonEle);
  // fireEvent.click(currentDom.getByText(/on/i));
  // console.log(currentDom.debug());
});
