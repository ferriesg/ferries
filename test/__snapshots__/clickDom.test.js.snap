// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click dom test 1`] = `"<clickDom />"`;
